### PR TITLE
Add support for preprocessed elevation data DTM and DOM from hoydedata.no

### PR DESCRIPTION
**Pull request by bgrotan**
Supported datums: UTM32,UTM33,UTM35
Supported elevation series: DTM and DOM for grids 50 meter, 10 meter and 1 meter

### CODE REVIEW

File: src/test/java/no/jskdata/HoydedataTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            }
+        });
+        assertEquals(1, fileNames.size());
+        h.clear();
+    }
+
+    public void testDownloadTwoDatasets() throws IOException {
+        Set<String> fileNames = new HashSet<>();
+        Hoydedata h = new Hoydedata();
+        h.setUtmzone("33");
```
Comment (halset) [inline src/test/java/no/jskdata/HoydedataTest.java]: Would it be possible to make it autoselect a utm zone if it is not specified?

File: src/main/java/no/jskdata/Hoydedata.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+/**
+ * Java wrapper around https://hoydedata.no/Laserinnsyn
+ */
+public class Hoydedata extends Downloader {
+
+    private String utmzone = "33";
+    private final Set<String> datasetIds = new HashSet<>();
+    private final String baseUrl = "https://hoydedata.no/LaserInnsyn/";
+
+    //private final List<String> datasets = new ArrayList<String>() {"DTM1","DOM1","DTM10","DOM10","DTM50","DOM50"};
```
Comment (halset) [inline src/main/java/no/jskdata/Hoydedata.java]: This can probably go away.

File: src/test/java/no/jskdata/HoydedataTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        List<File> files = h.getFilesforDataset("DTM1","33");
+        assertNotNull(files);
+        assertFalse(files.isEmpty());
+        assertTrue(files.size()>10);
+    }
+
+    public void testDownloadOneDataset() throws IOException {
+        Set<String> fileNames = new HashSet<>();
+        // test single file
+        fileNames.clear();
```
Comment (halset) [inline src/test/java/no/jskdata/HoydedataTest.java]: no need to clear a new `Set`

File: src/test/java/no/jskdata/HoydedataTest.java (hunk 1 above)
Comment (bgrotan) [reply src/test/java/no/jskdata/HoydedataTest.java]: UTM 33 would be a sane default as it contains the most data. 
The default is set as private String in Hoydedata.utmzone

**Comment by halset** [comment]
Nice, thanks! Let me know when hoydedata.no are able to serve files so we can test this all the way.

File: src/main/java/no/jskdata/Hoydedata.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    public void download(Receiver receiver) throws IOException {
+        for (String datasetId : datasetIds) {
+            if (receiver.shouldStop()) {
+                break;
+            }
+            List<File> datasetFiles = getFilesforDataset(datasetId,utmzone);
+            if (datasetFiles.isEmpty()) {
+                break;
```
Comment (halset) [inline src/main/java/no/jskdata/Hoydedata.java]: perhaps `continue;` instead of `break;`?

File: src/main/java/no/jskdata/Hoydedata.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            }
+            for (File file:datasetFiles) {
+                HttpURLConnection conn = openConnectionWithoutCookies(file.downloadUrl);
+                if (conn.getResponseCode() == 200) {
+                    receiver.receive(file.name, conn.getInputStream());
```
Comment (halset) [inline src/main/java/no/jskdata/Hoydedata.java]: Would it be possible to use the filename from the `Content-Disposition` http header?

Comment (halset) [reply src/main/java/no/jskdata/Hoydedata.java]: And as a consequence, perhaps getFilesforDataset, should return a urls instead of files.

**Pull request merged**